packages/@uppy/core/src/index.js

```diff
@@ -434,15 +455,8 @@ class Uppy {
   addFile (file) {
     const { files, allowNewUpload } = this.getState()
 
-    const onError = (msg) => {
-      const err = typeof msg === 'object' ? msg : new Error(msg)
-      this.log(err.message)
-      this.info(err.message, 'error', 5000)
-      throw err
-    }
-
     if (allowNewUpload === false) {
-      onError(new Error('Cannot add new files: already uploading.'))
+      this._showOrLogErrorAndThrow(new RestrictionError('Cannot add new files: already uploading.'))
```

Comment:
Should we emit `this.emit('restriction-failed', null, err)` on every RestrictionError, like we do in `upload()`?

Reply:
Great idea, added!